.ci/validation.sh

```diff
@@ -254,18 +254,19 @@ no-violation-test-configurate)
   ;;
 
 no-violation-test-josm)
-  CS_POM_VERSION="$(getCheckstylePomVersion)"
+  CS_POM_VERSION="10.21.2"
```

Comment:
we need snapshot version, as this is whole reason on why we run this job to compile JAR for PR and run this JAR in such project to see regression.

we have same for ANT project sample project
https://github.com/checkstyle/checkstyle/commit/5495106a50a0fdb3a4f9bee1280c14b4a5417dea#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R241-R243

recheck if jsom has https://github.com/sevntu-checkstyle/checkstyle-samples/blob/4c996615ed2582e9931968460638e19522215622/ant-project/ivysettings.xml#L5 

conceptually it is same goal of execution, but josm is real project , not a sample.

Reply:
by means of shell commands you need to udpate 
https://josm.openstreetmap.de/svn/trunk/ivysettings.xml

to add local repository as we have in checkstyle-examples.